src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1892,28 +1892,51 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
-                        //for each instancable item and add instance transforms.
-                        //If we have any line geometry that was not associated with an instance,
-                        //remove the previously added line data from the render package so the remaining lines can be added to the scene.
+                        //In the render package, all line geometry information (vertices, colors, indices) are stored together in the same arrays
+                        //regardless if the information is associated with an instancing transform or a line to be rendered directly to the scene.
+                        //If we are using IInstancingRenderPackage and there is vertex data associated with instancing transforms, then we need
+                        //handle the line data differently for each case.
+                        //For each instancable item, we need to create a unique Geometry3D object and add its instance transforms.
+                        //
+                        //Example: One Render package with a line, a rectangle (via instance transform) and another line
+                        //
+                        //Positions [P0,P1,P3,P3,P4,P5,P6,P7,P8] (line 1 (2 Positions), rectangle (5 Positions), line 2 (2 Positions))
+                        //Colors    [C0,C1,C2,C3,C4,C5,C6,C7,C8] (line 1 (2 Colors), rectangle (5 Colors), line 2 (2 Colors))
+                        //Indices [0,1,2,3,3,4,4,5,5,5,6,7,8] -> [(line 1) 1,2, (rectangle) 2,3,3,4,4,5,5,5,6 (line 2) 7,8]
+                        //
+                        //Note, the Indices array is always bound by the number of Positions. Values in Indices[] should not be less than 0 and should be less than count of Positions.
+                        //
+                        //In the example we have geometry data for the rectangle associated with an instance transform (ie rp.LineVertexRangesAssociatedWithInstancing.Any() == true)
+                        //LineVertexRangesAssociatedWithInstancing = {GuidForRectangle:(start:2,6)} -> meaning vertices 2-6 are associated with the rectangle.
+                        //
+                        //In Helix we need to create a unique Geometry3D object for the rectangle so that we can apply the instance transform.
+                        //This code will gather the data and transforms needed to call AddLineData for the rectangle.
                         if (rp.LineVertexRangesAssociatedWithInstancing.Any())
                         {
-                            //For each range of line vertices add the line data and instances to the scene
+                            //For each range of line vertices associated with an instance we will add the line data and instances to the scene
+                            //From the example above we would want to gather the data for the rectangle.
+                            //
+                            //For Positions [P0,P1,P3,P3,P4,P5,P6,P7,P8]
```

Comment:
Why is P3 twice? You mean P2, P3, right?

Reply:
Yes